docs/src/ref_impl/installer.md

```diff
@@ -5,67 +5,56 @@
-4. Write code for the installer in `ghaf.installer.installerCode`.
-
-```nix
-{config, ...}: {
-  ghaf.installer = {
-    enable = true;
-    imgModules = [
-      nixos-generators.nixosModules.raw-efi
-    ];
-    enabledModules = ["flushImage"];
-    installerCode = ''
-      echo "Starting flushing..."
-      if sudo dd if=${config.system.build.${config.formatAttr}} of=/dev/${config.ghaf.installer.installerModules.flushImage.providedVariables.deviceName} conv=sync bs=4K status=progress; then
-          sync
-          echo "Flushing finished successfully!"
-          echo "Now you can detach installation device and reboot to Ghaf."
-      else
-          echo "Some error occured during flushing process, exit code: $?."
-          exit
-      fi
-    '';
-  };
-}
-```
+Ghaf has a NixOS-like non-interactive, declarative installer instead of a
+conventional (imperative, non-declarative) installer as in most conventional
+Linux distributions. This is possible with the [Ghaf as
+Library](https://github.com/tiiuae/ghaf/pull/ghaf-based-project.md) approach:
+rather than clicking similar options during installation, you can configure the
+system once and deploy this configuration to the desired machines.
+
+
+To implement the pre-configured setting up approach, we used the
+[nixos-anywhere](https://github.com/nix-community/nixos-anywhere) tool.
+
+There are two separate ways to use it:
+* manually
+* with the Ghaf installation wizard.
+> NOTE: The Ghaf installation wizard is currently under development and cannot be used to create the required system configuration.
 
-After that you can build an installer image using this command:
+## Manual Installation
+
+To install Ghaf manually:
+
+1. Create your own flake using the Ghaf template:
 
 ```sh
-nix build .#nixosConfigurations.<CONFIGURATION>.config.system.build.installer
+nix flake init -t github:tiiuae/ghaf#target-x86_64-generic
 ```
 
-## Adding Installer Modules
+2. Edit it according to your preferences.
+
+3. Set the value of `ghaf.installer.sshKeys` to get an installer image.
 
-To add an installer module, replace the corresponding placeholders with your code and add this to your configuraiton:
+4. Build the installer image:
 
-```nix
-ghaf.installer.installerModules.<MODULE_NAME> = {
-  requestCode = ''
-    # Your request code written in Bash
-  '';
-  providedVariables = {
-    # Notice the dollar sign before the actual variable name in Bash.
-    <VARIABLE_NAME> = "$<VARIABLE_NAME>";
-  };
-};
+```sh
+nix build .#nixosConfigurations.PROJ_NAME-ghaf-debug.config.system.build.installer
 ```
 
-## Built-in Installer Modules
+5. Flash the installer image to your device:
+
+```sh
+sudo dd if=./result/iso/nixos-...-linux.iso of=/dev/YOUR_DEVICE conv=sync && sync
+```
 
-Provided variables show variable names in Nix. For actual names of variables in Bash, see the sources of the module.
+6. Run the image on the device.
 
-### flushImage
+7. Connect the device to the network using `wifi-connector`.
```

Comment:
Had to connect via Ethernet.

sudo wifi-connector wlp0s20f3 SSID PASSWORD
didn't work for me

It complains:
wlp0s4f0: error fetching interface information: Device not found

Is the wifi interface device hardcoded somewhere, or why it complains about different device I passed into the command?

Reply:
> Had to connect via Ethernet.
> 
> sudo wifi-connector wlp0s20f3 SSID PASSWORD didn't work for me
> 
> It complains: wlp0s4f0: error fetching interface information: Device not found
> 
> Is the wifi interface device hardcoded somewhere, or why it complains about different device I passed into the command?

To be honest - I did not test the `wifi-connector`. You might have found a bug. (I used [this](https://nixos.org/manual/nixos/stable/#sec-installation-manual-networking) with my smartphone hotspot.)

Reply:
Yes, this is bug, but [only in test of a connection](https://github.com/tiiuae/ghaf/pull/340/files#diff-2472af5c202571c85769038143336844721ff57b1f7882e16f21aa653ea39db7R40-R44). Although it shows that device not found, it should connect your device successfully if all input information is correct.

Reply:
> Yes, this is bug, but [only in test of a connection](https://github.com/tiiuae/ghaf/pull/340/files#diff-2472af5c202571c85769038143336844721ff57b1f7882e16f21aa653ea39db7R40-R44). 

The dependency indicates need to rebase this PR - the `wifi-connector` path in ghaf main has changed.
There's also a coming fix to the issue https://github.com/tiiuae/ghaf/pull/370/files

Reply:
This should be fixed in https://github.com/tiiuae/ghaf/pull/370